build/downloadTypos.js

```diff
@@ -0,0 +1,196 @@
+// @ts-check
+/**
+ * OSに合ったtyposのバイナリをダウンロードするスクリプト。
+ */
+const { exec } = require("child_process");
+const { promisify } = require("util");
+const { platform, arch } = require("os");
+const { join, resolve } = require("path");
+const {
+  mkdirSync,
+  existsSync,
+  unlinkSync,
+  createWriteStream,
+  rmSync,
+} = require("fs");
+const fetch = require("node-fetch");
+
+// OS名を定義するオブジェクト
+const OS = {
+  LINUX: "linux",
+  MACOS: "darwin",
+  WINDOWS: "win32",
+};
+// CPUアーキテクチャ名を定義するオブジェクト
+const CPU_ARCHITECTURE = {
+  X86_64: "x86_64",
+  ARM: "aarch64",
+};
+// 全バイナリのパス
+const BINARY_BASE_PATH = resolve(__dirname, "vendored");
+// typosのバイナリのパス
+const TYPOS_BINARY_PATH = resolve(BINARY_BASE_PATH, "typos");
+// 各OSとアーキテクチャに対応するtyposバイナリのダウンロード先URL
+const TYPOS_URLS = {
+  [OS.MACOS]: {
+    [CPU_ARCHITECTURE.ARM]:
+      "https://github.com/crate-ci/typos/releases/download/v1.21.0/typos-v1.21.0-aarch64-apple-darwin.tar.gz",
+    [CPU_ARCHITECTURE.X86_64]:
+      "https://github.com/crate-ci/typos/releases/download/v1.21.0/typos-v1.21.0-x86_64-apple-darwin.tar.gz",
+  },
+  [OS.LINUX]: {
+    [CPU_ARCHITECTURE.X86_64]:
+      "https://github.com/crate-ci/typos/releases/download/v1.21.0/typos-v1.21.0-x86_64-unknown-linux-musl.tar.gz",
+  },
+  [OS.WINDOWS]: {
+    [CPU_ARCHITECTURE.X86_64]:
+      "https://github.com/crate-ci/typos/releases/download/v1.21.0/typos-v1.21.0-x86_64-pc-windows-msvc.zip",
+  },
+};
+
+// 現在のOSとCPUアーキテクチャ
+const currentOS = platform();
+const currentCpuArchitecture =
+  arch() === "arm64" ? CPU_ARCHITECTURE.ARM : CPU_ARCHITECTURE.X86_64;
+// 7zバイナリのパス linuxとmacで異なるバイナリでないとエラーが出ることに注意
```

Comment:
こういうコメントの書き方テクニックもあったりします！
まあ特に何か変わるということは無いのですが、意図が伝わりやすいかも。
```suggestion
// 7zバイナリのパス
// WARNING: linuxとmacで異なるバイナリでないとエラーが出る
```